### PR TITLE
[FormRecognizer] Implemented Reading Order feature for content recognition

### LINKED ISSUE #19807: [FormRecognizer] Add support for the new Reading Order feature for content recognition

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/azure/azure-sdk-for-net/issues/19807

- Added new property to `RecognizeContentOptions`.
- Updated docstrings.
- Added tests and recordings.

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 1]
```diff
@@ -515,6 +515,35 @@ public void StartRecognizeContentWithNoSupporttedLanguage()
...
+            var naturalOrderFormPage = naturalOrderOperation.Value.Single();
+
+            ValidateFormPage(basicOrderFormPage, includeFieldElements: true, expectedPageNumber: 1);
+            ValidateFormPage(naturalOrderFormPage, includeFieldElements: true, expectedPageNumber: 1);
+
+            var basicOrderLines = basicOrderFormPage.Lines.Select(f => f.Text);
+            var naturalOrderLines = naturalOrderFormPage.Lines.Select(f => f.Text);
+
+            CollectionAssert.AreEquivalent(basicOrderLines, naturalOrderLines);
+            CollectionAssert.AreNotEqual(basicOrderLines, naturalOrderLines);
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:544]: I think it's not worth testing whether the returned lines in fact follow the `Basic` or the `Natural` ordering algorithm (checking bounding boxes, checking proximity of related text lines, etc.). To test the SDK, simply checking that changing the specified order updated the ordering behavior in the service seems enough.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormPage.cs [hunk 2]
```diff
@@ -85,11 +85,12 @@ internal FormPage(int pageNumber, float width, float height, float textAngle, Le
...
-        /// proximity is treated with higher priority. As the sorting order depends on the detected text, it may change across
-        /// images and OCR version updates. Thus, business logic should be built upon the actual line location instead of order.
+        /// When 'IncludeFieldElements' is set to <c>true</c>, a list of recognized lines of text. An empty list otherwise.
+        /// The maximum number of lines returned is 300 per page. The lines are sorted top to bottom, left to right, although
+        /// in certain cases proximity is treated with higher priority. As the sorting order depends on the detected text, it
+        /// may change across images and OCR version updates. Thus, business logic should be built upon the actual line location
+        /// instead of order. For calls to recognize content, this list is always populated and the sorting algorithm for the
+        /// returned lines can be changed with the option <see cref="RecognizeContentOptions.ReadingOrder"/>.
         /// </summary>
         public IReadOnlyList<FormLine> Lines { get; }
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormPage.cs:95]: to make sure I understood, there is no default ReadingOrder applied, right?

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormPage.cs:95]: We're not setting anything to be the default on the client side, but if we don't send anything, the service will use the `Basic` algorithm.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
-    // Setting the "internal" access modifier while we don't expose the reading order
-    // feature as part of our design.
+using Azure.Core;
 
-    internal enum ReadingOrder
+namespace Azure.AI.FormRecognizer
+{
+    /// <summary>
+    /// Specifies the order in which recognized text lines are returned in calls to
+    /// recognize content. This can be set in the <see cref="RecognizeContentOptions"/>.
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:10]: The plan is to have ReadingOrder in every endpoint, so maybe we should remove the `recognize content` part

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
     {
+        /// <summary>
+        /// The lines are sorted top to bottom, left to right, although in certain cases
+        /// proximity is treated with higher priority.
+        /// </summary>
         Basic,
+
+        /// <summary>
+        /// The lines are returned in a more human-friendly reading order. Related text lines
+        /// are likely to be in sequence.
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:26]: I don't feel comfortable generalizing about `human-friendly reading order` without knowing if it actually includes everyone and if it adapts according to the language.
The service also explains it like `.  The new algorithm uses positional information to keep nearby lines together` so maybe taking something from there might be bette?

**Review by maririos — COMMENTED** [review]
some docstring comments

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs (hunk 4 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:26]: Thank you for pointing this out. If we find out it doesn't include all FR supported languages, we may need to update it in the [documentation](https://docs.microsoft.com/en-us/azure/cognitive-services/form-recognizer/whats-new#march-2021) as well.

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:26]: for sure!

Comment (witemple-msft) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:26]: The text I used was "uses heuristics to mimic the way a human reader would read the document." Maybe we should workshop this briefly to figure out the right way to communicate the difference between the basic and natural algorithms.

**Review by witemple-msft — APPROVED** [review]
LGTM, concerns about how to describe "natural" reading order notwithstanding.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs:24]: Could either of you start a conversation in our Teams chat so we can get all languages in agreement on what the docstring should be?

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/ReadingOrder.cs]: Updated to `The algorithm uses positional information to keep nearby lines together`.

**Pull request merged**